miserable pull request

42LayerToiletpaper opened the pull request:
## Description
added thirdparty library simde and making it work in Base64 file.
Attempted to declare function in Base64 using simde, but that causes errors. its 1 am, so well call it a day and proceed with the pr.

<!-- Please include a summary of the change and which issue is fixed here. -->

## Checklist
- [x ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [x ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

Austernpilz commented on src/openms/include/OpenMS/FORMAT/Base64.h:
```diff
@@ -165,6 +168,23 @@ namespace OpenMS
       UInt32 i;
     };
 
+    
+    const simde__m128i mask1 = simde_mm_set1_epi32(0x3F000000);//00111111 00000000 00000000 00000000
+    const simde__m128i mask2 = simde_mm_set1_epi32(0x003F0000);//00000000 00111111 00000000 00000000
+    const simde__m128i mask3 = simde_mm_set1_epi32(0x00003F00);//00000000 00000000 00111111 00000000
+    const simde__m128i mask4 = simde_mm_set1_epi32(0x0000003F);//00000000 00000000 00000000 00111111
+
+
+
+    const simde__m128i allA = simde_mm_set1_epi8('A');
+    const simde__m128i alla = simde_mm_set1_epi8('a' - 26);
+    const simde__m128i all0 = simde_mm_set1_epi8('0' - 52);
+    const simde__m128i allPlus = simde_mm_set1_epi8('+');
+    const simde__m128i allSlash = simde_mm_set1_epi8('/');
+
+    const simde__m128i shuffleMask = simde_mm_setr_epi8(2, 2, 1, 0, 5, 5, 4, 3, 8, 8, 7, 6, 11, 11, 10, 9);
+    const simde__m128i shuffleMask2 = simde_mm_setr_epi8(3,2,1,0,7,6,5,4,11,10,9,8,15,14,13,12);
+
```
What happens here?

the 4 Blocks of Code for a Mask ok, 
themn you Add Letters and Numbers, but why the + / ?? 
and why just an 'A'

the last step? completly mysterious

Austernpilz commented on src/openms/include/OpenMS/FORMAT/Base64.h:
```diff
@@ -205,7 +227,36 @@ namespace OpenMS
+    data = simde_mm_shuffle_epi8(data, shuffleMask2);
+
+
+    simde__m128i bigLetterMask= simde_mm_cmplt_epi8(data, _mm_set1_epi8(26)); // (a < b) ? 0xFF : 0x00
```
maybe it's the Simdi encoding, but what does this comment mean?

Austernpilz commented on src/tests/class_tests/openms/source/AAIndex_test.cpp:
```diff
@@ -36,271 +36,34 @@
-  TEST_REAL_SIMILAR(AAIndex::acidic('D'),1.0)
-  TEST_REAL_SIMILAR(AAIndex::acidic('A'),0.0)
-END_SECTION
+// should be : "eJxLTEpOSU1Lz8jMys7JzcsvKCwqLiktK6+orGJwdHJ2cXVz9/D08vbx9fMPCAwKDgkNC4+IjIpiMDQyNjE1M7ewNGAAACpTFWY="
```
you delete so much code and comments, can you provide more Info
Also are you still testing the same things?

Austernpilz commented on src/openms/include/OpenMS/FORMAT/Base64.h:
```diff
@@ -49,6 +49,9 @@
+#include <../thirdparty/simde/simde/x86/sse.h>
 
 #include <QByteArray>
 #include <zlib.h>
```
lexicographical sort?

Austernpilz commented:
For this big include as the thrirdpartylibrary, some form of message would be helpful.
Also in the "modify cmake file and note down what needs to be changed partially"
most comments don't explain so much, there just there and new. Or did I miss something?

timosachsenberg committed:
fix error related to empty spectra in bruker files (#6897)

Co-authored-by: timosachsenberg <sachsenb@ibminode02.Cs.Uni-Tuebingen.De>

timosachsenberg committed:
fix error related to empty spectra in bruker files (#6897) (#6898)

Co-authored-by: timosachsenberg <sachsenb@ibminode02.Cs.Uni-Tuebingen.De>

KyowonJeong committed:
Stable version - all scoring functions fixed. Ready to do DL

jpfeuffer committed:
[fix][pyopenms] adhoc sign libs that are copied into wheel (#6904)

jpfeuffer committed:
[pyopenms] add jenkinsfile for silicon build

jpfeuffer committed:
agent

jpfeuffer committed:
agent

jpfeuffer committed:
agent

jpfeuffer committed:
checkout

jpfeuffer committed:
steps

jpfeuffer committed:
steps

jpfeuffer committed:
path

jpfeuffer committed:
contrib

jpfeuffer committed:
submodules

jpfeuffer committed:
copmpiler

jpfeuffer committed:
compiler

jpfeuffer committed:
delete

jpfeuffer committed:
delete all

jpfeuffer committed:
steps?

jpfeuffer committed:
conda

jpfeuffer committed:
zsh?

jpfeuffer committed:
init?

jpfeuffer committed:
source conda init

jpfeuffer committed:
qt5dir

jpfeuffer committed:
creds

jpfeuffer committed:
fix error in test

jpfeuffer committed:
activate conda again

jpfeuffer committed:
Update Jenkinsfile

timosachsenberg committed:
Update Jenkinsfile (#6906)

jpfeuffer committed:
fix some py scripts

jpfeuffer committed:
make similar to GH actions

jpfeuffer committed:
no conda needed for twine. installed with brew now

jpfeuffer committed:
escape dollar

jpfeuffer committed:
single quotes

jpfeuffer committed:
conda activate

KyowonJeong committed:
Dummy for FDR estimation is optimized. Get ready for PR. First we will build this version for release.

jpfeuffer committed:
complete release branch

jpfeuffer committed:
forgot to remove

jpfeuffer committed:
Merge pull request #6905 from OpenMS/pyopenmsJenkinsfile

Pyopenms jenkinsfile (for macOS silicon)

jpfeuffer committed:
Update Jenkinsfile

jpfeuffer committed:
Merge pull request #6908 from OpenMS/pyopenmsJenkinsfile

Update pyoms Jenkinsfile such that it checks out into a local branch with the same name

KyowonJeong committed:
refactoring FLASHDeconvAlgorithm

KyowonJeong committed:
Base version. Need to implement negative isotope intensities.

KyowonJeong committed:
Negative isotope implemented!!! Now let's go with refactoring.

KyowonJeong committed:
A bit of refactoring done. Build verions

KyowonJeong committed:
A bit of refactoring done. Build version. Windows error fixed.

KyowonJeong committed:
A bit of refactoring done. Build version. Windows error fixed2.

KyowonJeong committed:
A bit of refactoring done. Build version. Windows error fixed3.

cbielow committed:
fix bogus icon message on Linux (where we do not embed icons)

timosachsenberg committed:
Merge branch 'develop' into RDB-logs

cbielow committed:
Merge branch 'develop' into fix_msg_icon

cbielow committed:
Update src/openms/source/CHEMISTRY/RibonucleotideDB.cpp

cbielow committed:
Update src/openms/source/CHEMISTRY/RibonucleotideDB.cpp

cbielow committed:
Update src/openms/source/CHEMISTRY/RibonucleotideDB.cpp

cbielow committed:
Merge pull request #6884 from OpenMS/RDB-logs

Better logging in RibonucleotideDB.cpp

cbielow committed:
Merge pull request #6909 from cbielow/fix_msg_icon

fix bogus icon message on Linux (where we do not embed icons)

SGSSGene committed:
Feature - reading parameters via JSON file (#6879)

* fix-test: let us close that CTD stream before checking

* feat: adds parameter import via JSON

This allows to specify a JSON file, instead of a ini/ctd file as an
input to `-ini`. This allows workflows (e.g. CWL) to produce parameter
files that can be consumed by OpenMS tools.

About the Json input files:
    These file are a dictionary of key to values (scalar). Only the special case
    of files will require a map.

A small example file for SeedListGenerator:
```
{
    "in": {
        "class": "File",
        "path": "topp/PepXMLFile_test.mzML"
    },
    "test": true,
    "out_prefix": "test_cwl_",
    "threads": 1
}
````

Some programs have nested parameter, in this case two different key
synatxes can be used. Either keys with `algorithm:threshold` as OpenMS
also uses internally or `algorithm__threshold`. The second version is
required for external tools that don't handle ':' very well or have
special meanings.
e.g.:
```
{
...
    "algorithm:threshold": 10,
    "algorithm__type": "xml",
...
}
```

* Test for inbound JSON + -write_ini

* fix: json, unit test and ignore outdated parameter

* fix: adjusting accordingly to review

* fix: only parse JSON if ending is .json

* patch: ParamCWLFile::load indicates success of loading

* patch: adding two unit tests for JSON ini files

* Revert "Test for inbound JSON + -write_ini"

This reverts commit c6b911ad3c4c58a8311b8e65f81bd273575b1b1e.

* Update src/openms/include/OpenMS/FORMAT/ParamCWLFile.h

Co-authored-by: Chris Bielow <chris.bielow@fu-berlin.de>

* fix: throw if file couldn't be opened

* Apply suggestions from code review

Co-authored-by: Chris Bielow <chris.bielow@fu-berlin.de>

* patch: ParamCWLFile::load is now static

* fix: add class test for ParamCWLFile

* Update src/tests/class_tests/openms/source/ParamCWLFile_test.cpp

* fix linter warning

---------

Co-authored-by: Michael R. Crusoe <michael.crusoe@gmail.com>
Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>
Co-authored-by: Chris Bielow <chris.bielow@fu-berlin.de>

cbielow committed:
Fix slow string conversion (#6910)

* fix very inefficient string conversion

* remove unused function

timosachsenberg committed:
Merge pull request #6911 from OpenMS/develop

update release branch with latest changes

dependabot[bot] committed:
Bump src/pyOpenMS/pyopenms-docs from `8c1f161` to `577aca4`

Bumps [src/pyOpenMS/pyopenms-docs](https://github.com/OpenMS/pyopenms-docs) from `8c1f161` to `577aca4`.
- [Release notes](https://github.com/OpenMS/pyopenms-docs/releases)
- [Commits](https://github.com/OpenMS/pyopenms-docs/compare/8c1f16113f36b5de98001fef2f39ea75f984d07f...577aca40cec7eba1158127e3d42bdd8e1464a7f6)

---
updated-dependencies:
- dependency-name: src/pyOpenMS/pyopenms-docs
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

timosachsenberg committed:
Merge pull request #6912 from OpenMS/dependabot/submodules/src/pyOpenMS/pyopenms-docs-577aca4

Bump src/pyOpenMS/pyopenms-docs from `8c1f161` to `577aca4`

poshul committed:
Update IsobaricQuantifier.cpp

Set intensity < 1e-11 to count channel as empty

poshul committed:
Update MRMTransitionGroupPicker.h

Fixes issues where behavior diverges between architectures due to floating poin weirdness

timosachsenberg committed:
Update src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h

poshul committed:
Update MRMTransitionGroupPicker.h

Fixed unintended consequences of cat-keyboard interaction

poshul committed:
Update MRMTransitionGroupPicker.h

Accumulate should start from 0.0

poshul committed:
Update MS3TMT10Plex_test.consensusXML

Update test

poshul committed:
Update TMTTenPlexMethod_test.consensusXML

poshul committed:
Update MS3TMT10Plex_test.consensusXML

timosachsenberg committed:
Merge branch 'develop' into release/3.0.0

timosachsenberg committed:
Merge pull request #6914 from OpenMS/fix_floatingpoint_IsobaricQuant

Update IsobaricQuantifier.cpp

timosachsenberg committed:
Merge pull request #6916 from OpenMS/fixMRMfloatingpt

Update MRMTransitionGroupPicker.h

KyowonJeong committed:
A few minor bugs fixed. And functions have been refactored. A bit of documentation added.

KyowonJeong committed:
A few minor bugs fixed. And functions have been refactored. A bit of documentation added + one possible bug with static varialbe fixed...

JeeH-K committed:
Merge branch 'develop' into FLASHDeconvFix

timosachsenberg committed:
Update MSFraggerAdapter.cpp

timosachsenberg committed:
Merge pull request #6917 from OpenMS/timosachsenberg-patch-1

remove unicode character from description in MSFraggerAdapter.cpp

timosachsenberg committed:
Merge pull request #6883 from JeeH-K/FLASHDeconvFix

[Fix] bug fixing on FLASHDeconv

timosachsenberg committed:
Merge remote-tracking branch 'origin/develop' into release/3.0.0

jpfeuffer committed:
ignore Cython warnings for py 3.8

timosachsenberg committed:
Update update_year_number.yml

timosachsenberg committed:
Merge pull request #6920 from OpenMS/jpfeuffer-patch-1

ignore Cython warnings for py 3.8

timosachsenberg committed:
Merge pull request #6921 from OpenMS/run-update-year-number-once-per-year

Update update_year_number.yml

timosachsenberg committed:
Update update_year_number.yml

timosachsenberg committed:
Merge pull request #6924 from OpenMS/timosachsenberg-patch-1

Update update_year_number.yml

timosachsenberg committed:
Update update_year_number.yml

timosachsenberg committed:
Merge pull request #6925 from OpenMS/timosachsenberg-patch-1

Update update_year_number.yml

timosachsenberg committed:
Updated year number

timosachsenberg committed:
Merge pull request #6926 from OpenMS/update_year_number

[ByGitAction] Updated year number in the header of files

cbielow committed:
fix license year

cbielow committed:
allow external JSON target / update nlohmann::json to proper lib

cbielow committed:
minor cleanup

cbielow committed:
fix find_package to propagate results to parent; a bit more verbosity

cbielow committed:
also update LICENSE.md

cbielow committed:
also update LICENSE.md

cbielow committed:
Merge pull request #6928 from cbielow/feat_external_libs

Support external libs (JSON, SQLiteCpp)

jpfeuffer committed:
Merge pull request #6930 from OpenMS/develop

update release RC from develop

poshul committed:
Merge branch 'develop' into license_year

cbielow committed:
Merge pull request #6929 from OpenMS/license_year

also update LICENSE.md

poshul committed:
Update refman_overwrite.tex.in

We don't need to define Bbbk any more and newtxtext is not in the livetex-basic package, so lets get rid of it

poshul committed:
Merge pull request #6933 from OpenMS/Remove-dependency-on-newtxtext

Update refman_overwrite.tex.in

poshul committed:
remove old tutorial

cbielow committed:
fix minimal cmake version

cbielow committed:
fix minimal cmake version

poshul committed:
remove refs to tutorial

poshul committed:
Merge remote-tracking branch 'upstream/develop' into removetutorial

cbielow committed:
Update doc/doxygen/install/install-macs.doxygen

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

cbielow committed:
Update doc/doxygen/install/install-macs.doxygen

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

cbielow committed:
Merge pull request #6935 from cbielow/feat_external_libs

fix minimal cmake version (3.24)

poshul committed:
Merge pull request #6936 from poshul/removetutorial

Remove pdf tutorial from targets

Commit:
update mascot test

Commit:
update mascot test

timosachsenberg committed:
Merge pull request #6938 from OpenMS/fix/mascot

update mascot test

timosachsenberg committed:
Merge pull request #6943 from cbielow/r3.0.0_up

update Release3.0.0

jpfeuffer committed:
Update cmake in pyopenms-wheels.yml

fixes #6945

timosachsenberg committed:
Update CHANGELOG_PARAMS

jpfeuffer committed:
Update knime_tests.yml

jpfeuffer committed:
Update knime_tests.yml

jpfeuffer committed:
last fixes for vcpkg.
- allow coin includes in different subdir
- remove boost test (just because)
- list subset of boost deps

jpfeuffer committed:
fix coin include dir cmake logic

jpfeuffer committed:
typo

jpfeuffer committed:
fix conditional find pkg

jpfeuffer committed:
comment fix conditional find pkg

jpfeuffer committed:
Merge pull request #6877 from OpenMS/feat/brewVcpkgCMake

[FEATURE] Make CMake work with coin from brew. Complete the vcpkg recipe.

jpfeuffer committed:
Update docs with warning in FloatDataArray.pyx

timosachsenberg committed:
Merge pull request #6947 from OpenMS/jpfeuffer-patch-7

Update docs with warning in FloatDataArray.pyx

PierreSnell committed:
feat: :sparkles: Added ion mobility possibility for dataframe (#6941)

* feat: :sparkles: Added ion mobility possibility for dataframe

* Update MSExperiment.h

* Update src/openms/include/OpenMS/KERNEL/MSExperiment.h

* Update src/openms/include/OpenMS/KERNEL/MSExperiment.h

Co-authored-by: Julianus Pfeuffer <pfeuffer@informatik.uni-tuebingen.de>

* Update src/openms/include/OpenMS/KERNEL/MSExperiment.h

Co-authored-by: Julianus Pfeuffer <pfeuffer@informatik.uni-tuebingen.de>

* fixes

* fix

* add getSpectrum to AreaIterator

* added IM data point

* check if IM data arrays present

* add for second verson as well

* doc & default parameter for _dataframe

* Update src/pyOpenMS/pyopenms/_dataframes.py

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

* added mzml and test

* misc fix

* better test

* Changed for IM and make test pass

* fix test file

* fix extraction by avoiding temporary data view

---------

Co-authored-by: Julianus Pfeuffer <pfeuffer@informatik.uni-tuebingen.de>
Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

timosachsenberg committed:
Release/3.0.0 (#6948)

* change context

* change context

* change context

* change context

* change context

* change context

* change context

* push release contrib container

* proper container name

* fix error related to empty spectra in bruker files (#6897) (#6898)

Co-authored-by: timosachsenberg <sachsenb@ibminode02.Cs.Uni-Tuebingen.De>

* also update LICENSE.md

* fix minimal cmake version

* Update doc/doxygen/install/install-macs.doxygen

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

* update mascot test

* Update CHANGELOG_PARAMS

---------

Co-authored-by: timosachsenberg <sachsenb@ibminode02.Cs.Uni-Tuebingen.De>
Co-authored-by: Julianus Pfeuffer <pfeuffer@informatik.uni-tuebingen.de>
Co-authored-by: Chris Bielow <chris.bielow@fu-berlin.de>

timosachsenberg committed:
Merge pull request #6949 from OpenMS/develop

update release branch

SGSSGene committed:
fix: json reader entries for output files, should be strings

SGSSGene committed:
Satisfy linter

SGSSGene committed:
test: add test to check for correct input file type

SGSSGene committed:
test: fix JSON-ini reader based tool tests

SGSSGene committed:
feat| JSON-ini reader, evaluate is_executable tag, to allow file and string as input

SGSSGene committed:
apply lint suggestion

SGSSGene committed:
fix: pass through 'is_executable' on input files

timosachsenberg committed:
SimpleSVM_test_speedup

timosachsenberg committed:
Merge pull request #6951 from OpenMS/SimpleSVM_test_speedup

SimpleSVM test speedup

poshul committed:
Merge branch 'develop' into fix/json_reader_output_files_shall_be_string

dependabot[bot] committed:
Bump THIRDPARTY from `e215209` to `0dda83f`

Bumps [THIRDPARTY](https://github.com/OpenMS/THIRDPARTY) from `e215209` to `0dda83f`.
- [Release notes](https://github.com/OpenMS/THIRDPARTY/releases)
- [Commits](https://github.com/OpenMS/THIRDPARTY/compare/e2152090f83531096ab771b608d561e56325d2c5...0dda83f91697960235b4cb33a448c4f40c553e08)

---
updated-dependencies:
- dependency-name: THIRDPARTY
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

poshul committed:
Merge pull request #6950 from SGSSGene/fix/json_reader_output_files_shall_be_string

fix: json reader entries for output files, should be strings

timosachsenberg committed:
Merge pull request #6957 from OpenMS/release/3.0.0

Merge back Release/3.0.0

jpfeuffer committed:
Update update_version_numbers.yml

jpfeuffer committed:
Update update_version_numbers.yml

jpfeuffer committed:
Update update_version_numbers.yml

jpfeuffer committed:
Update update_version_numbers.yml

jpfeuffer committed:
Updated OpenMS package version number

jpfeuffer committed:
Update CHANGELOG

jpfeuffer committed:
Merge pull request #6739 from OpenMS/update_package_number

[ByGHAction] Updated OpenMS package version numbers

jpfeuffer committed:
Update update_version_numbers.yml

cbielow committed:
fixes https://github.com/OpenMS/OpenMS/issues/6959

cbielow committed:
apply to Clang in general, not only Apple

jpfeuffer committed:
Unify action trigger for container and wheels

timosachsenberg committed:
Merge pull request #6961 from OpenMS/jpfeuffer-patch-1

Unify action trigger for container and wheels

timosachsenberg committed:
Merge pull request #6960 from cbielow/fix_FMA_apple

fixes failing tests on Mac Silicon

timosachsenberg committed:
Update openms-ci.yml

timosachsenberg committed:
Update openms-ci.yml

timosachsenberg committed:
Update openms-ci.yml

cbielow committed:
Revert "Update IsobaricQuantifier.cpp"

timosachsenberg committed:
Update .github/workflows/openms-ci.yml

jpfeuffer committed:
Update openms-ci.yml

jpfeuffer committed:
try 5.10.8

jpfeuffer committed:
5.15.2 is the latest available

jpfeuffer committed:
Merge pull request #6965 from OpenMS/timosachsenberg-patch-1

Update openms-ci.yml

timosachsenberg committed:
Merge pull request #6966 from OpenMS/revert-6914-fix_floatingpoint_IsobaricQuant

Revert "Update IsobaricQuantifier.cpp"

cbielow committed:
Merge branch 'develop' of https://github.com/OpenMS/OpenMS into HEAD

cbielow committed:
reverse logic of #6877, to allow pre-configured OpenMS source builds to run without weird error messages

cbielow committed:
fix some compile errors for Win and minor optimizations

cbielow committed:
fix for clang/gcc

cbielow committed:
useAVX2 on MSVC by default

cbielow committed:
clean up SIMD compile flags

jpfeuffer committed:
Replace #6974

jpfeuffer committed:
Replace #6974

cbielow committed:
Merge pull request #6975 from OpenMS/changeCoinFind

[CMAKE] detect coin-or in different places

cbielow committed:
Revert "reverse logic of #6877, to allow pre-configured OpenMS source builds to run without weird error messages"

This reverts commit 30e5def115eea2f71b5703db7596c61a2a6ffb3f.

cbielow committed:
Merge branch 'develop' of https://github.com/OpenMS/OpenMS into base64_simd

cbielow committed:
cleaning up a bit

cbielow committed:
noop

cbielow committed:
document AVX requirement

cbielow committed:
extend test

cbielow committed:
update PR number

cbielow committed:
switch to using a library-interface for SIMDe to avoid `...thirdparty\simde\simde\x86\sse.h(3363,39): error C4100: 'i': unreferenced formal parameter ` on MSVC due to external includes since we are using /we4100 and /we4189

radusuciu committed:
Add some project urls to pyopenms package

radusuciu committed:
Fix link in pyopenms package description

dependabot[bot] committed:
Bump src/pyOpenMS/pyopenms-docs from `577aca4` to `04c8016`

Bumps [src/pyOpenMS/pyopenms-docs](https://github.com/OpenMS/pyopenms-docs) from `577aca4` to `04c8016`.
- [Release notes](https://github.com/OpenMS/pyopenms-docs/releases)
- [Commits](https://github.com/OpenMS/pyopenms-docs/compare/577aca40cec7eba1158127e3d42bdd8e1464a7f6...04c8016d6dd22cf38b71ce6374b22bf445100f26)

---
updated-dependencies:
- dependency-name: src/pyOpenMS/pyopenms-docs
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

timosachsenberg committed:
Merge pull request #6981 from radusuciu/fix_pyopenms_description_link

Fix link in pyopenms package description

poshul committed:
Merge branch 'develop' into dependabot/submodules/THIRDPARTY-0dda83f

poshul committed:
Update CHANGELOG

[nop] correct release date for 3.0

timosachsenberg committed:
Update src/pyOpenMS/setup.py

Co-authored-by: Julianus Pfeuffer <pfeuffer@informatik.uni-tuebingen.de>

poshul committed:
Merge branch 'develop' into dependabot/submodules/src/pyOpenMS/pyopenms-docs-04c8016

timosachsenberg committed:
Merge pull request #6984 from OpenMS/poshul-patch-1

Update CHANGELOG

jpfeuffer committed:
Merge pull request #6980 from radusuciu/add_pyopenms_project_urls

Add some project urls to pyopenms package

cbielow committed:
add FindModule for SIMDe

cbielow committed:
fix a few compile warnings

timosachsenberg committed:
update pxds for cython 3 compatibility (#6985)

* update pxds for cython 3 compatibility

* preserve spaces

* preserve spaces

cbielow committed:
move SIMDe code to cpp

poshul committed:
Merge pull request #6955 from OpenMS/dependabot/submodules/THIRDPARTY-0dda83f

Bump THIRDPARTY from `e215209` to `0dda83f`

cbielow committed:
make SIMDe includes 'external' i.e. they are not affected by 'C4100: 'i': unreferenced formal parameter'

cbielow committed:
better button text to avoid confusion with loading data files

cbielow committed:
Merge pull request #6988 from cbielow/toppas_wording

TOPPAS: better button text to avoid confusion with loading data files

timosachsenberg committed:
[CLEANUP] Remove first set of unused tools (#6963)

* remove some files

* remove InclusionExclusionListCreator

* remoe InclusionExclusionListCreator

* removed SvmTheoreticalSpectrumGenerator

* remove another test

* forgot paramhandlerdocumenter

* update docs

* some more doc updates

* some more fixes

* update changelog

* forgot paramhandlerdocumenter

update docs

some more doc updates

* some more fixes

update changelog

cbielow committed:
switch to target_include_directories

jpfeuffer committed:
Merge pull request #6983 from OpenMS/dependabot/submodules/src/pyOpenMS/pyopenms-docs-04c8016

Bump src/pyOpenMS/pyopenms-docs from `577aca4` to `04c8016`

timosachsenberg committed:
Remove_oldsvm (#6990)

* Removed RT and PT predict model

* remove old SVM wrapper code

* update changelog

* remove RTEvaluation and update changelog

* Update CHANGELOG

dependabot[bot] committed:
Bump THIRDPARTY from `0dda83f` to `402992f` (#6993)

Bumps [THIRDPARTY](https://github.com/OpenMS/THIRDPARTY) from `0dda83f` to `402992f`.
- [Release notes](https://github.com/OpenMS/THIRDPARTY/releases)
- [Commits](https://github.com/OpenMS/THIRDPARTY/compare/0dda83f91697960235b4cb33a448c4f40c553e08...402992fbb05ac567fb6c5ba4f3dbac679eaf5649)

---
updated-dependencies:
- dependency-name: THIRDPARTY
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

cbielow committed:
NOOP

cbielow committed:
add support for endianess when decoding; remove bogus testcase

timosachsenberg committed:
[Cleanup] Remove some of the eval tools (#6998)

* removed eval tools

* update changelog

* nop

* forgot executable

timosachsenberg committed:
[Cleanup] Remove PepNovoAdapter (#6997)

* remove PepNovoAdapter

* clean up non-existing tool entries

* fix test

* Update CHANGELOG

cbielow committed:
Merge branch 'develop' into base64_simd

cbielow committed:
fix compile error

cbielow committed:
Merge branch 'base64_simd' of https://github.com/cbielow/OpenMS into base64_simd

timosachsenberg committed:
[cleanup] remove CompNovo (#6999)

* remove CompNovo

* changelog

* removed additioal files

* Update CHANGELOG

cbielow committed:
Merge branch 'develop' into base64_simd

cbielow committed:
fix compile errors

cbielow committed:
fix warning

cbielow committed:
fix wrong base64 data in test

cbielow committed:
fix bug

jpfeuffer committed:
Get rid of dependency on C compilers (#7002)

cbielow committed:
fix bug in compression length

cbielow committed:
remove useless assign

cbielow committed:
[DOC] for internals; and saving an include

cbielow committed:
move compression code to a better-tested place

cbielow committed:
Merge branch 'develop' into base64_simd

cbielow committed:
update simde

cbielow committed:
remove harmful include

cbielow committed:
fix test

timosachsenberg committed:
remove tool also from KNIME packaging (#7004)

updates knime_package_support.cmake 
apparently had no effect that the tool was missing so this is just a cleanup

timosachsenberg committed:
Update openms_clang_format.yml (#7005)

* Update openms_clang_format.yml

* Update openms_clang_format.yml

timosachsenberg committed:
[discussion] remove utils and model as topp tool with utilities category (#7001)

* move utils to topp tool (adapt category)

* move utils to topp folder and change category

* fix

* fix test

* update TOPP documenter

* readd GUI utils

* more documentation updates

* update documentation

* fix boost dependency

* fixed comment in build script

* Update Tutorial_Final.cpp

fix typo

---------

Co-authored-by: Samuel Wein <sam@samwein.com>

cbielow committed:
fix warning

cbielow committed:
fix bug in Class_test which displays bogus error message even when test passed

cbielow committed:
Merge pull request #6978 from cbielow/base64_simd

[FEATURE] Base64 encoding/decoding using SIMDe

timosachsenberg committed:
Sage MS (#6786)

* reading of version numbers

* scaffold for PercolatorInfile::load

* nop

* minimal parsing

* wip

* more entries

* nop

* Update PercolatorInfile.cpp

* add modification parsing to pin file

* use proper score column for main score

* Update src/openms/include/OpenMS/FORMAT/PercolatorInfile.h

* creates basic test file

* update test

* update test

* update pin file parsing

* Apply suggestions from code review

* some minor work on adapter

* support multi file pin files

* nop

* improved sage support

* improved sage support

* support for multiple files

* add sage subscores for percolator / mokapot rescoring

* support for subscores propagation to percolator

* some preparation to for config template

* wrap easy parameters

* change map_inde to id_merge_index, add enzyme support

* untested support for all relevant parameters

* small refactoring

* first working version

* Update src/topp/SageAdapter.cpp

* Apply suggestions from code review

Co-authored-by: Julianus Pfeuffer <pfeuffer@informatik.uni-tuebingen.de>

* expand search meta data

* prepration for thirdparty test

* removed unneded copy

* fix some missing update error

* add precursor RT and MZ to id results

* fix wrong paramter in test. good test data still WIP

* add Sage test

* mend

* Update src/topp/SageAdapter.cpp

* update test filename in idXML

* Update third_party_tests.cmake

---------

Co-authored-by: timosachsenberg <sachsenb@ibminode02.Cs.Uni-Tuebingen.De>
Co-authored-by: timosachsenberg <sachsenb@ibminode04.cs.uni-tuebingen.de>
Co-authored-by: Samuel Wein <sam@samwein.com>
Co-authored-by: Julianus Pfeuffer <pfeuffer@informatik.uni-tuebingen.de>

cbielow committed:
[FIX] mzML threading issue  (#7011)

* remove dependency on member variable which is being accessed by multiple threads (race condition)

* add changelog

timosachsenberg committed:
Fix_sage (#7013)

* make idxml user parameter output deterministic

* new fix

* added sage to  changelog

* update sage output

timosachsenberg committed:
remove errorstring from XMLHandler.pxd (#7014)

timosachsenberg committed:
[cleanup] remove linear mapping PIP (#7016)

* remove linear mapping PIP

* remove tutorial and test

* update some cmake files

---------

Co-authored-by: timosachsenberg <sachsenb@ibminode02.Cs.Uni-Tuebingen.De>

cbielow committed:
fix neon flags (not needed)

timosachsenberg committed:
[cleanup] remove DocumentIDTagger and IDPool from share (#7015)

* remove DocumentIDTagger and IDPool from share

* fix cmake files

* ammend

---------

Co-authored-by: timosachsenberg <sachsenb@ibminode02.Cs.Uni-Tuebingen.De>

cbielow committed:
Merge pull request #7017 from cbielow/fix_neon

fix neon flags (not needed)

matteopilz committed:
fixed broken / deprecated link (#7018)

timosachsenberg committed:
removed EnzymaticDigestionLogModel (#7019)

cbielow committed:
fix compile error on MacOS-ARM

cbielow committed:
code to detect ISA extensions

cbielow committed:
add Changelog

cbielow committed:
Merge branch 'develop' into fix_neon

cbielow committed:
fix unreferenced formal param

cbielow committed:
Merge branch 'fix_neon' of https://github.com/cbielow/OpenMS into fix_neon

cbielow committed:
Merge pull request #7022 from cbielow/fix_neon

[FEATURE] Report active ISA extensions
